packages/components/addon/components/hds/tabs/index.js

```diff
@@ -0,0 +1,83 @@
+import Component from '@glimmer/component';
+import { A } from '@ember/array';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class HdsTabsIndexComponent extends Component {
+  @tracked tabNodes = A([]);
+  @tracked tabIds = A([]);
+  @tracked panelNodes = A([]);
+  @tracked panelIds = A([]);
+  @tracked selectedIndex = 0; // isSelected, 0 = default
+
+  @action
+  didInsertTab(tabId, isSelected, element) {
+    this.tabNodes = A([...this.tabNodes, element]);
+    this.tabIds = A([...this.tabIds, tabId]);
+
+    if (isSelected) {
+      this.setSelectedTabIndex(this.tabIds.length - 1);
+    }
```

Comment:
There is something that makes me nervous here: essentially we're counting on the fact that the DOM node are already there, already rendered/available, but we're not 100% sure in reality they're there. what if we move this logic of activating the selected one only on the `didInsert` of the main `tabs` component? in that case we're 100% sure all the children are there.

the logic would be:
- each `tab` and `panel` on "insert" reports its presence to the `tabs`
- the tabs creates the arrays with the references for `tab[]` and `panel[]` (elements and/or ids) plus stores a reference to the "selected" index
- on insert of `tabs` it triggers a `click` on the `tab` corresponding to the selected index
this should work, and maybe simplify the logic too?

Reply:
I refactored to get rid of this. I'm instead using a data attribute to store the initial tab and then setting the initial tab index on insert of the tabs (parent) component now. 

Don't know if this is better but let me know what you think. (I wanted to try to avoid attaching/setting html attributes from within the JavaScript vs. within the HBS template like you did in your branch just because to me it seems clearer to keep those in the templates.)

Reply:
I'm going to go ahead and resolve this comment but please open a new one if you want me to make additional changes.